### ISSUE TITLE
Make re-running installer act like other Windows Installers?

### ISSUE THREAD

**Issue by hatton** [Squirrel/Squirrel.Windows#317]
Thinking about what happens when a user runs the installer for a version that he/she already has installed.

Currently, if the user runs a full installer that finds it has nothing to do, Squirrel just tries to run the application. 

Why would the user do this?  Here's what occurs to me:

### By accident, because they think this is how you're supposed to run the program, or because they forgot that it was installed
In this case, it doesn't matter much what we do; the current behavior isn't too bad; probably better to though to give them an indication that they already have run the installer and that they already have this version.

### Because they need to re-install
We had a case recently where a user's antivirus quarantined one of the components of our application, making it unusable.  We've also had cases of failed in-app updates, also possibly caused by antivirus. In these  cases, the user could have found out how to uninstall, then run the installer. But a more normal Windows thing to do is just to run the installer again, expecting that you'll get some UI giving you some choices.

### Because they can't find the shortcut anymore
Probably the simplest thing is to do a repair install, as above.

### Because they want to uninstall
Some installers offer this, so a user might be trying to use this approach to uninstall something.

We might be able to do this work. Thoughts?

**Comment by paulcbetts** [comment]
> Currently, if the user runs a full installer that finds it has nothing to do, Squirrel just tries to run the application.

I'd tend to agree, I've been meaning to change this behavior for awhile now

> Because they need to re-install

I think this is by-far the most common reason, I think this is what we'll end up doing (i.e. flat-out pave the existing install directory and recreate it)

**Comment by hatton** [comment]
This has gotten worse. Currently, it does not run the app, it crashes with this error:

![image](https://cloud.githubusercontent.com/assets/8448/7972566/ba7211bc-0a0a-11e5-87c0-d8cce434675d.png)

And there is no install log to read.

@paulcbetts Our team needs to work on this right now. Please let us know if you have any advice before we start on it. Thx.

**Comment by paulcbetts** [comment]
Have you checked in `%LocalAppData%\SquirrelTemp`, that's where the initial log shows up

**Comment by hatton** [comment]
Was looking in the wrong place:

```
2015-06-03 17:24:07> Program: Starting Squirrel Updater: --install .
2015-06-03 17:24:07> Program: Starting install, writing to C:\Users\john\AppData\Local\SquirrelTemp
2015-06-03 17:24:07> Program: About to install to: C:\Users\john\AppData\Local\BloomAlpha
2015-06-03 17:24:07> CheckForUpdateImpl: Reading RELEASES file from C:\Users\john\AppData\Local\SquirrelTemp
2015-06-03 17:24:08> Unhandled exception: System.AggregateException: One or more errors occurred. ---> System.DivideByZeroException: Attempted to divide by zero.
   at Squirrel.UpdateManager.ApplyReleasesImpl.<createFullPackagesFromDeltas>d__ab.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__67.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Update.Program.<Install>d__31.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Squirrel.Update.Program.executeCommandLine(String[] args)
   at Squirrel.Update.Program.main(String[] args)
---> (Inner Exception #0) System.DivideByZeroException: Attempted to divide by zero.
   at Squirrel.UpdateManager.ApplyReleasesImpl.<createFullPackagesFromDeltas>d__ab.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<ApplyReleases>d__67.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.<FullInstall>d__f.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Update.Program.<Install>d__31.MoveNext()<---

**Comment by paulcbetts** [comment]
@hatton Rebuild Squirrel from source and drop it in your tools dir, not seeing where we divide by zero here

**Comment by hatton** [comment]
Sorry, I didn't realize our fork was active in this area (I'm not coding on it). We've fixed the divide by zero. @JohnThomson has also committed the repair-on-rerun, and we'll make a PR here after some testing of that.

**Comment by JohnThomson** [comment]
On another PR, I've repaired a bug in your attempt to display an "Open log file" button in that dialog

**Comment by hatton** [comment]
We've done this on our fork, but need to get a PR together. It's working well.

**Comment by lipis** [comment]
:palm_tree:

**Comment by paulcbetts** [comment]
This is already fixed, we now kill processes belonging to your app on install and uninstall
